### ISSUE TITLE
Path param regex requires excessive backslash escaping

### ISSUE THREAD

**Issue by metacity** [ninjaframework/ninja#188]
Hi,

When using path parameters with regex matching (á la `/assets/{fileName: .*}`), Ninja seems to be requiring an extra "layer" of escaping for backslashes. So for example when setting up a route to match a numeric id:

``` java
// Does NOT work
router.GET().route("/customers/{id: \\d+}").with(Controller.class, "method");

// WORKS
router.GET().route("/customers/{id: \\\\d+}").with(Controller.class, "method");

// Also works
router.GET().route("/customers/{id: [0-9]+}").with(Controller.class, "method");
```

Regex without a named path parameter seems to work like intended. I'll take a closer look later, but after a quick glance through the routing related classes, could this be a result of first extracting the regex part of the route, converting it to a string and then compiling it to `Pattern`? If someone has an obvious solution, go ahead :)

Of course this could just be a documented feature.. Although matching a literal "\" would become quite tedious. Thoughts?

**Comment by raphaelbauer** [comment]
Good catch. Actually I also have to look into the code to see what is happening... Any improvements all insights of course appreciated :)

**Comment by bazi** [comment]
Resolved in [PR#247](https://github.com/ninjaframework/ninja/pull/247). Should it be closed?

@raphaelbauer  Who closes issues? Or the author @metacity ?

**Comment by metacity** [comment]
Ah yes, thank you! :+1:
